tapfreighter/chain_porter.go

```diff
@@ -607,14 +607,43 @@ func (p *ChainPorter) transferReceiverProof(pkg *sendPackage) error {
 		log.Debugf("Attempting to deliver proof for script key %x",
 			key.SerializeCompressed())
 
+		proofCourierAddr, err := proof.ParseCourierAddrString(
+			string(out.ProofCourierAddr),
+		)
+		if err != nil {
+			return fmt.Errorf("failed to parse proof courier "+
+				"address: %w", err)
+		}
+
+		// Initiate proof courier service handle from the proof
+		// courier address found in the Tap address.
+		courier, err := proofCourierAddr.NewCourier(
+			ctx, p.cfg.ProofCourierCfg,
+		)
+		if err != nil {
+			return fmt.Errorf("unable to initiate proof courier "+
+				"service handle: %w", err)
+		}
+
+		// Update courier events subscribers before attempting to
+		// deliver proof.
+		p.subscriberMtx.Lock()
+		courier.SetSubscribers(p.subscribers)
```

Comment:
Why wasn't this needed before?

Reply:
Courier subscribers were set before in `ChainPorter.RegisterSubscriber`. That setup made sense when there was a single courier instance tide to a variable in the `ChainPorter` struct. But now that we're creating and destroying a courier handle within `ChainPorter.transferReceiverProof` we don't have a chance to hook into `ChainPorter.RegisterSubscriber`.